### PR TITLE
Issue/#54 comment schema

### PR DESCRIPTION
**Pull request by rodgergit**
## Proposed Changes

  - Added comment schema
  - Added comment property to post schema
  - Added api call for comments which creates a comment and then updates the post schema with the comment id

### CODE REVIEW

**Comment by rodgergit** [comment]
Unit testing in progress.

File: server/test/integration/post.spec.js [hunk 1]
```diff
@@ -192,4 +192,18 @@ describe('Posts API', () => {
     expect(response1.status).toBe(404);
     done();
   });
+
+  // it('tests the comment method and makes sure it appends to post schema ', async done => {
+  //   const commentData = {
+  //     title: 'Test comment',
```
Comment (kaplunket) [inline server/test/integration/post.spec.js]: Comments don't have a title?

File: server/src/routes/posts.js [hunk 2]
```diff
@@ -52,4 +53,22 @@ router.delete('/:id', async (req, res) => {
...
+// Commenting once and updating post schema with comment id
+// eslint-disable-next-line no-unused-vars
+router.post('/:id/comment', async (req, res) => {
+  const comment = new Comment({
+    comment_id: req.body.comment_id,
+    body: req.body.body,
+  });
+  try {
+    const newComment = await comment.save();
+    res.status(201).json(newComment);
```
Comment (GrunWong) [inline server/src/routes/posts.js]: Returning newComment in the response isn't a part of the API, if it's possible could this be corrected to reflect the API documentation.

Comment (GrunWong) [reply server/src/routes/posts.js]: Also I believe the response should be sent as the last line of the 'try' statement.

File: server/src/routes/posts.js [hunk 3]
```diff
@@ -52,4 +53,22 @@ router.delete('/:id', async (req, res) => {
...
+
+    await Post.update({ _id: req.params.id }, { $push: { comment_id: newComment._id } });
+    res.status(200).send();
```
Comment (GrunWong) [inline server/src/routes/posts.js]: I don't think we should be sending 2 successful responses, this line might not be necessary?

File: server/src/routes/posts.js [hunk 4]
```diff
@@ -52,4 +53,22 @@ router.delete('/:id', async (req, res) => {
...
+  } catch (err) {
+    res.status(404).json({ message: err.message });
```
Comment (GrunWong) [inline server/src/routes/posts.js]: Is 404 the right status code to send back in this case? I believe, it actually should be 400.

File: server/src/routes/posts.js (hunk 2 above)
Comment (kaplunket) [reply server/src/routes/posts.js]: should this also go after the update post?

Comment (rodgergit) [reply server/src/routes/posts.js]: How would commnets be stored to the db then?

Comment (GrunWong) [reply server/src/routes/posts.js]: Line 64, where you call comment.save() is when it actually saves into the DB.

**Pull request merged**